tds.go

```diff
@@ -1201,43 +1238,47 @@ initiate_connection:
 		return nil, err
 	}
 
-	if encrypt != encryptNotSup {
-		var config *tls.Config
-		if pc := p.TLSConfig; pc != nil {
-			config = pc
-			if config.DynamicRecordSizingDisabled == false {
-				config = config.Clone()
+	if p.Encryption != msdsn.EncryptionStrict {
```

Comment:
> p.Encryption != msdsn.EncryptionStrict  [](http://example.com/codeflow?start=3&length=40)

instead of checking p, how about checking whether `outputbuf.transport` is already encrypted?

Reply:
Added flag to check if outputbuf.transport is encrypted.  Also checked if the config p is being modified anywhere but that doesn't seem to be the case.